File: csharp/src/Training.Workshop/M6/S06Builder/Start/Hall.cs
using System.Text;

namespace Training.Workshop.M6.S06Builder.Start;

/// <summary>Start - mutowalny węzeł: sala z listą seansów.</summary>
public sealed class Hall
{
    private readonly string _name;
    private readonly List<Screening> _screenings = [];

    public Hall(string name)
    {
        _name = name;
    }

    public void Add(Screening screening)
    {
        _screenings.Add(screening);
    }

    public int Size()
    {
        return _screenings.Count;
    }

    public string Render()
    {
        var text = new StringBuilder(_name).Append('\n');
        if (_screenings.Count == 0)
        {
            text.Append("  (brak seansow)\n");
        }
        _screenings.ForEach(screening => text.Append(screening.Render()));
        return text.ToString();
    }
}

File: typescript/src/workshop/m6/s06_builder/start/Hall.ts
import type { Screening } from './Screening.js';

/** Start - mutowalny węzeł: sala z listą seansów. */
export class Hall {
  private readonly screenings: Screening[] = [];

  constructor(private readonly name: string) {}

  add(screening: Screening): void {
    this.screenings.push(screening);
  }

  size(): number {
    return this.screenings.length;
  }

  render(): string {
    let text = `${this.name}\n`;
    if (this.screenings.length === 0) {
      text += '  (brak seansow)\n';
    }
    this.screenings.forEach((screening) => {
      text += screening.render();
    });
    return text;
  }
}
